subt/doc/virtual-howto.md

```diff
@@ -56,50 +55,59 @@ It is necessary to kill & start again both sim and bridge docker for each run. N
 solution not covered by this HOWTO.
 
 
-## Build image for local testing
+## Working with the main "robotika" image
+
+### Build
 ```
-cd subt/docker
-./build.bash sln_bash
-./run.bash sln_bash
+./subt/docker/build.bash robotika
 ```
 
-Now you are in the docker so you can update current OSGAR:
-```
-cd osgar
-git pull
-cd ..
-osgar/subt/script/sync.sh
+### Run locally
+
+```commandline
+./subt/docker/run.bash robotika
 ```
-And run robotika solution code:
+which by default runs `./osgar/subt/docker/robotika/run_solution.bash` inside the container.
+To get a shell inside the docker instead, run
+```commandline
+./subt/docker/run.bash robotika bash
 ```
-osgar/subt/docker/robotika/run_solution.bash
+and you can call `run_solution.bash` when you are ready.
+
+A copy of `osgar` directory from the time of the build of the image is located
+at `/home/developer/subt_solution/osgar`. For local development it is advantageous
+to mount your `osgar` directory from the host over this directory in the container.
+
+```commandline
+DEVEL=1 ./subt/docker/run.bash robotika bash
 ```
 
+When you do so, you can edit the files as you are used to and also log files
+from the runs are saved to the host and not to the container. To rebuild the ROS
+nodes call `./osgar/subt/script/sync.sh`, to run the solution call 
+`./osgar/subt/docker/robotika/run_solution.bash`.
```

Comment:
Should I run the the two last commands from inside docker, or from outside? The "mount directory from host" part leaves me confused if we are still in the "mounted directory" mode of work, or back in the previous "inside container" one.

Reply:
For the two modes to be transparent, all commands must be always run inside the container. So nothing is ever run outside of the container.

There are two main reasons/advantages to using the mount:
1. you can use editing tools you generally use outside of the container (for example I use pycharm)
2. you can use the git setup you already have (you don't have to setup git again inside of the container)

Reply:
I have updated the description to note that the commands should be run inside of the container.